R/edit.R

```diff
@@ -1,3 +1,39 @@
+#' Open file for editing
+#'
+#' Opens a file for editing in RStudio, if that is the active environment, or
+#' via [utils::file.edit()] otherwise. If the file does not exist, it is
+#' created. If the parent directory does not exist, it is also created.
+#'
+#' @param path Path to target file.
+#'
+#' @return Target path, invisibly.
+#' @export
+#' @keywords internal
+#'
+#' @examples
+#' \dontrun{
+#' edit_file("DESCRIPTION")
+#' edit_file("~/.gitconfig")
+#' }
+edit_file <- function(path) {
+  path <- user_path_prep(path)
+  dir_create(path_dir(path), recursive = TRUE)
```

Comment:
Should this call `use_directory()`? The additional messaging seems like it would be nice.

Reply:
Hmm, I'm a bit boxed into a corner atm. `use_directory()` assumes that input `path` is relative to the active project. Whereas all the `write_*()` helpers and `edit_file()` assume the input path can be taken at face value.

I'll take this as evidence that the functions re: directory creation aren't quite right yet. Will work on that.

Reply:
Oh good point; can definitely happen after this PR.

Reply:
Will sort this out here: https://github.com/r-lib/usethis/issues/393